src/main/java/frc/robot/motors/TalonFXWrapper.java

```diff
@@ -0,0 +1,141 @@
+package frc.robot.motors;
+import static edu.wpi.first.units.Units.Amps;
+import static edu.wpi.first.units.Units.RPM;
+import static edu.wpi.first.units.Units.Rotations;
+import static edu.wpi.first.units.Units.Volts;
+import static java.lang.Math.PI;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+
+/**
+ * TalonFx wrapper class, using DC motor sim + sim states.
+ */
+public class TalonFXWrapper extends TalonFX implements LoggedMotor {
+
+	// Attributes
+	private static final double K_GEAR_RATIO = 10.0;
+	private static final double INERTIA_CONSTANT = 0.001;
+
+	// Components
+	private final DCMotor gearbox;
+	// Sim model for calculations
+	// getSimState() acccesses the default sim
+	private final DCMotorSim motorSimModel;
+
+	public static final double LOOP_PERIOD_MS = 0.020;
+
+	/**
+	 * Constructor with device ID.
+	 * @param deviceId the CAN ID of the motor
+	 */
+	public TalonFXWrapper(int deviceId) {
+		this(deviceId, DCMotor.getKrakenX60(1));
+	}
+
+	/**
+	 * Constructor with device ID and motor type.
+	 * @param deviceId the CAN ID of the motor
+	 * @param motorType the motor type
+	 */
+	public TalonFXWrapper(int deviceId, DCMotor motorType) {
```

Comment:
Can we choose a name for the DCMotor across all the constructors and member variables for consistency? Right now we have gearbox vs. motorType vs. dcMotor.